sklearn/metrics/pairwise.py

```diff
@@ -330,12 +339,9 @@ def pairwise_distances_argmin_min(X, Y, axis=1, metric="euclidean",
     sklearn.metrics.pairwise_distances
     sklearn.metrics.pairwise_distances_argmin
     """
-    dist_func = None
-    if metric in PAIRWISE_DISTANCE_FUNCTIONS:
-        dist_func = PAIRWISE_DISTANCE_FUNCTIONS[metric]
-    elif not callable(metric) and not isinstance(metric, str):
-        raise ValueError("'metric' must be a string or a callable")
-
+    if batch_size is not None:
+        warnings.warn("'batch_size' was deprecated in version 0.19 and will "
+                      "be removed in version 0.21.", DeprecationWarning)
```

Comment:
You should set `block_size` according to `batch_size`, i.e. `block_size = int(ceil(batch_size ** 2 / BYTES_PER_FLOAT / 1024 / 1024))` (I think).